src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockIndexCreator.java

```diff
@@ -0,0 +1,95 @@
+package ee.tuleva.onboarding.comparisons.fundvalue.retrieval;
+
+import ee.tuleva.onboarding.comparisons.fundvalue.FundValue;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
+
+@Slf4j
+@Service
+public class GlobalStockIndexCreator implements ComparisonIndexRetriever {
+    public static final String KEY = "NEW_GLOBAL_STOCK_INDEX";
+
+    NamedParameterJdbcTemplate jdbcTemplate;
```

Comment:
Separation of concerns. Use the Repository design pattern to encapsulate & separate the data access logic from other higher-abstraction-level business logic in this class. 

https://martinfowler.com/eaaCatalog/repository.html

Reply:
In other words, this class should depend on the `FundValueRepository` (which, in turn, depends on the `jdbcTemplate`) not directly on the `jdbcTemplate`.